test/unit/StakingPool/processExpirations.js

```diff
@@ -108,4 +117,269 @@ describe('processExpirations', function () {
+
+    await stakingPool.processExpirations(true);
+
+    await increaseTime(TRANCHE_DURATION * 8);
+
+    // expire all tranches
+    await stakingPool.processExpirations(true);
+
+    // Validate tranches are expired
+    for (let i = 0; i < tranches.length; i++) {
+      const tranche = tranches[i];
+      const expiredTranche = await stakingPool.expiredTranches(tranche);
+
+      const activeDepositsAtTranche = maxTranche - tranche + 1;
+
+      expect(expiredTranche.accNxmPerRewardShareAtExpiry).to.gt(0);
+      expect(expiredTranche.stakeAmountAtExpiry).to.equal(amount.mul(activeDepositsAtTranche));
+      expect(expiredTranche.stakeShareSupplyAtExpiry).to.equal(baseStakeShares * activeDepositsAtTranche);
+
+      const trancheData = await stakingPool.tranches(tranche);
+      expect(trancheData.stakeShares).to.equal(0);
+      expect(trancheData.rewardsShares).to.equal(0);
+    }
+
+    // Validate globals active stake, stake share supply and rewards shares supply
+    {
+      const activeStake = await stakingPool.activeStake();
+      const stakeSharesSupply = await stakingPool.stakeSharesSupply();
+      const rewardsSharesSupply = await stakingPool.rewardsSharesSupply();
+
+      expect(activeStake).to.equal(0);
+      expect(stakeSharesSupply).to.equal(0);
+      expect(rewardsSharesSupply).to.equal(0);
+    }
+  });
+
+  it('Expires buckets updating rewards per second and lastAccNxmUpdate', async function () {
+    const { stakingPool } = this;
+    const {
+      members: [user],
+    } = this.accounts;
+
+    const { amount, tokenId, destination } = processExpirationsFixture;
+
+    const { firstActiveTrancheId } = await getTranches();
+
+    await stakingPool.connect(user).depositTo([
+      {
+        amount,
+        trancheId: firstActiveTrancheId,
+        tokenId,
+        destination,
+      },
+    ]);
+
+    await generateRewards(stakingPool, this.coverSigner, daysToSeconds(10), 0);
+
+    const accNxmPerRewardsShareBefore = await stakingPool.accNxmPerRewardsShare();
+    const rewardPerSecondBefore = await stakingPool.rewardPerSecond();
+    const lastAccNxmUpdateBefore = await stakingPool.lastAccNxmUpdate();
+
+    await increaseTime(BUCKET_DURATION);
+
+    await stakingPool.processExpirations(false);
+
+    const firstActiveBucketId = await stakingPool.firstActiveBucketId();
+    const accNxmPerRewardsShareAfter = await stakingPool.accNxmPerRewardsShare();
+    const rewardPerSecondAfter = await stakingPool.rewardPerSecond();
+    const lastAccNxmUpdateAfter = await stakingPool.lastAccNxmUpdate();
+    const expiredBucketRewards = await stakingPool.rewardBuckets(firstActiveBucketId);
+    const rewardsSharesSupply = await stakingPool.rewardsSharesSupply();
+
+    const bucketStartTime = firstActiveBucketId.mul(BUCKET_DURATION);
+    const elapsed = bucketStartTime.sub(lastAccNxmUpdateBefore);
+
+    expect(expiredBucketRewards).to.gt(0);
```

Comment:
can we have an exact value here for `expiredBucketRewards` ?

strengthen the assertion